### ISSUE TITLE
Wrong FormatStyle.MEDIUM Date formatting for Norwegian Locale

### ISSUE THREAD

**Issue by mkotsbak** [cquiroz/scala-java-locales#54]
it seems like the date formatting for Norwegian is not working properly. Should be like German (28.06.2016), but I get: "2016 28" (only year and day of month).

Code is:

```
LocaleRegistry.installLocale(nb_NO)
Locale.setDefault(Locale.forLanguageTag("nb_NO"))
Date.format(DateTimeFormatter.ofLocalizedDate(FormatStyle.MEDIUM))
```

**Comment by cquiroz** [comment]
The call `Locale.forLanguageTag(tag)` takes a BCP47 tag which should in this case be `nb-NO` rather than `nb_NO`. The variable for `installLocale` uses `nb_NO` because it is scala code, and it needs a valid identifier

The java API is very non-restrictive, you are allowed to pass anything to `Locale.forLanguageTag` and it will try to produce a locale as best as possible. In this case it will go back to the root locale. Ideally it should throw an exception instead but the API documentation and the JVM implementation lets you pass anything

```
    LocaleRegistry.installLocale(nb_NO)

    Locale.setDefault(Locale.forLanguageTag("nb_NO"))
    // Produces y MMM d   
    DateFormat.getDateInstance(DateFormat.MEDIUM).asInstanceOf[SimpleDateFormat].toPattern())

    Locale.setDefault(Locale.forLanguageTag("nb-NO"))    
    // Produces d. MMM y
    DateFormat.getDateInstance(DateFormat.MEDIUM).asInstanceOf[SimpleDateFormat].toPattern()
```

In summary, convert `Locale.forLanguageTag("nb_NO")` to `Locale.forLanguageTag("nb-NO")`

**Comment by mkotsbak** [comment]
It works fine with "nb-NO" yes. Maybe we should have a test that ensures that invalid language tag gives the same result as with JVM?

**Comment by cquiroz** [comment]
Makes sense, there is no documentation at all on how the JVM creates `Locale` for unknown tags. This is very annoying as I bet many people will make the same mistake. The function ideally would be `Locale.forLanguageTag():Option[Locale]` to only return valid locales

**Comment by mkotsbak** [comment]
Yes, given the same underscore format used also in eg.  bash shell. Strange that it does not throw one of the exceptions Java loves. Maybe we could print a warning to stderr?

**Comment by cquiroz** [comment]
I don't know a good solution for this without breaking the API, printing a message to stderr or log sounds too intrusive for my taste

**Comment by mkotsbak** [comment]
Well, but it only would happen if doing what should be similar to writing a syntax error, but yes, maybe someone autodetects a new locale from somewhere that is not yet supported.
